modules/charon-core/src/main/java/org/wso2/charon3/core/attributes/ComplexAttribute.java

```diff
@@ -58,7 +60,7 @@ public void setSubAttributesList(Map<String, Attribute> subAttributesList) {
      * @param attributeName
      * @return Attribute
      */
-    public Attribute getSubAttribute(String attributeName) throws CharonException {
+    public Attribute getSubAttribute(String attributeName) {
```

Comment:
We are not recommending to change public methods as we need to preserve the backward compatibility as these public methods can be used from different other places.